lib/src/base_client.dart

```diff
@@ -153,20 +153,23 @@ abstract class BaseClient implements Client {
     var request = new Request(method, url);
 
     if (headers != null) request.headers.addAll(headers);
-    if (encoding != null) request.encoding = encoding;
-    if (body != null) {
-      if (body is String) {
-        request.body = body;
-      } else if (body is List) {
-        request.bodyBytes = body.cast<int>();
-      } else if (body is Map) {
-        request.bodyFields = body.cast<String, String>();
-      } else {
-        throw new ArgumentError('Invalid request body "$body".');
+   if(encoding != null) request.encoding = encoding; else {
```

Comment:
We will need to run `dartfmt` and the body of the `if` needs to be wrapped in `{}`.